### PR TITLE
Teach flutter how to avoid the iOS keyboard

### PR DESCRIPTION
**Pull request by abarth**
In the future, we'll probably want to plumb through the keyboard animation
parameters, but for now we match the behavior on Android where we move the
widgets out of the way of the keyboard without animation.

### CODE REVIEW

**Comment by abarth** [comment]
@chinmaygarde 

I wasn't able to test this change locally because I couldn't get a locally build engine to run on iOS simulation.  I'd like some help with that before landing this patch.  Thanks!

File: sky/shell/platform/ios/sky_surface.mm [hunk 1]
```diff
@@ -112,6 +115,14 @@ - (instancetype)initWithShellView:(sky::shell::ShellView*)shellView {
 
     _shell_view.reset(shellView);
     self.multipleTouchEnabled = YES;
+
+    [[NSNotificationCenter defaultCenter] addObserver:self
```
Comment (chinmaygarde) [inline sky/shell/platform/ios/sky_surface.mm]: Generally a good practice to get rid of these registrations (and collect the viewport metrics instance above) in the dealloc method. I realize the surface is around for a long time though.

Comment (abarth) [inline sky/shell/platform/ios/sky_surface.mm]: Will do.

sky::SkyEnginePtr has a destructor that frees the ViewportMetrics.  Do we need to do more work for the viewport metrics or does dealloc call the C++ destructors?

**Comment by chinmaygarde** [comment]
To test locally on a simulator, there is a Flutter directory in the build artifacts generated by ninja that contains an Xcode project. In that project is a Local.xcconfig. you can point the variables in that config file to your Dart project and use the project normally. I will document this in the engine wiki.

File: sky/shell/platform/ios/sky_surface.mm (hunk 1 above)
Comment (chinmaygarde) [inline sky/shell/platform/ios/sky_surface.mm]: C++ destructors don't need to be called manually.

**Comment by abarth** [comment]
I tried that with https://github.com/flutter/engine/wiki/Flutter-Apps-on-iOS#3-running-your-flutter-project but it had trouble during startup.  It told me that runFromBundle (or whatever) had a null string and failed validation.

File: sky/shell/platform/ios/sky_surface.mm (hunk 1 above)
Comment (abarth) [inline sky/shell/platform/ios/sky_surface.mm]: Done.

**Comment by chinmaygarde** [comment]
lgtm. I am investigating startup issue now.

**Comment by abarth** [comment]
Yay, ran simulator and verified that this works.

**Comment by chinmaygarde** [comment]
lgtm

**Pull request merged**